src/components/view/Details.js

```diff
@@ -64,16 +109,15 @@ var Details = React.createClass({
           <h2>You have 3 days and 4 hours to get your task done.</h2>
         </div>
         <div className="content-wrapper">
-          <a className="pictogram-icon wundercon icon-checkbox gray mr1"></a>
-          <h2 className="inline-block m0 mb1 main-task">{task.title}</h2>
+          <TaskInlineEdit className={classList} textClasses="main-task mb1 inline-block" onClick={self.completeMainTask} updateValue={self.updateTaskTitle} title={task.title}/>
```

Comment:
Why not let the component handled update the value itself?  Why would this container need to know about syncing out the title, if it can't directly update it itself?